### PR TITLE
Fixed impl

### PR DESCRIPTION
**Pull request by da-steve101**
An implementation of Fixed for chisel3.
Just transforms over SInt so shouldn't affect FIRRTL

### CODE REVIEW

**Comment by da-steve101** [comment]
As a side note, perhaps Bits should not be sealed? I can understand UInt/SInt/Bool but I think Bits would commonly want to be extended ( custom floating point implementations etc ). Fixed could even go in a separate chisel-utils library if this was the case. I think Fixed should be in chisel but it is an option worth considering.

**Comment by da-steve101** [comment]
This is currently broken

**Comment by da-steve101** [comment]
Seems to be working ok. Still not sure if fixed should be put into firrtl or just SInt based implementation as  here. Tend towards layer on SInt as it is essentially the same. Changed the fromBits method to be a connect instead of ":=". I think fromBits should force a connection.

File: chiselFrontend/src/main/scala/chisel3/core/Bits.scala [hunk 1]
```diff
@@ -213,7 +213,7 @@ sealed abstract class Bits(dirArg: Direction, width: Width, override val litArg:
   /** Reinterpret cast to an UInt.
     *
     * @note value not guaranteed to be preserved: for example, a SInt of width
-    * 3 and value -1 (0b111) would become an UInt with value 7
```
Comment (aswaterman) [inline chiselFrontend/src/main/scala/chisel3/core/Bits.scala]: "0b" here means "base 2," not a width.

**Comment by aswaterman** [comment]
If it's possible to implement it w/o changing FIRRTL, it seems simpler not to change FIRRTL.

Relatedly, can we do this as a library on top of Chisel, instead of making it part of Chisel?

**Comment by ucbjrl** [comment]
We're doing similar development for Chisel DSP support. Stay tuned.

**Comment by da-steve101** [comment]
@aswaterman You can make it external to chisel, but would require changing a few things. Main one is Bits not being sealed (which I really think it shouldn't be). I would also be in favour of unsealing UInt/SInt/Bool but the argument isn't as strong.

**Comment by chick** [comment]
@da-steve101 have you taken a look at the FixedPoint type now baked into firrtl and chisel3 and used extensively in dsptools.  I think it covers everything that is in here.

**Comment by ducky64** [comment]
Superseded by FixedPoint

**Pull request closed without merging**